copy/copy.go

```diff
@@ -79,8 +80,10 @@ func Copy(ctx context.Context, srcRoot, src, dstRoot, dst string, opts ...Opt) e
 		if err != nil {
 			return err
 		}
-		if err := MkdirAll(ensureDstPath, 0755, ci.Chown, ci.Utime); err != nil {
+		if createdDirs, err := MkdirAll(ensureDstPath, 0755, ci.Chown, ci.Utime); err != nil {
 			return err
+		} else {
+			defer fixCreatedParentDirs(createdDirs, ci.Utime)
```

Comment:
related to https://github.com/docker/buildx/pull/2745#discussion_r1810862520, maybe it could also apply ownership as well:

```suggestion
			defer fixCreatedParentDirs(createdDirs, ci.Chown, ci.Utime)
```